classes/class-newsmatchdonation-shortcode.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * The class for the News Match Donation Shortcode, and associated functions
+ *
+ * @package NewsMatchDonation\Shortcode
+ */
+
+/**
+ * Register a shortcode
+ */
+class NewsMatchDonation_Shortcode {
+	/**
+	 * The prefix used for this plugin's options saved in the options table
+	 *
+	 * @var string $options_prefix The prefix for this plugin's options saved in the options table
+	 */
+	protected $option_prefix = '';
+
+	/**
+	 * The constructor
+	 */
+	public function __construct() {
+		$this->option_prefix = NewsMatchDonation_Settings::$options_prefix;
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'register_assets' ) );
+		add_shortcode( 'newsmatch_donation_form', array( $this, 'donation_form_shortcode' ) );
+	}
+
+	/**
+	 * Define the donation form shortcode.
+	 * <p>
+	 *     This shortcode is intended for use somewhere on a page.
+	 * </p>
+	 * <p>
+	 *     Example usage:
+	 *     Add donation form with no Salesforce campaign id and no default donation amount specified:
+	 *     [newsmatch_donation_form]
+	 *
+	 *      Add donation form with no Salesforce campaign id and $50.00 as the default donation amount:
+	 *      [newsmatch_donation_form amount="50"]
+	 *
+	 *      Add a donation form with a Salesforce campaign id of "foo" and $25.00 as the default donation amount:
+	 *      [newsmatch_donation_form sf_campaign_id="foo" amount="25"]
+	 *
+	 *      Add a donation form with a Salesforce campaign id of "foo" and do not specify a default donation amount:
+	 *      [newsmatch_donation_form sf_campaign_id="foo"]
+	 * </p>
+	 *
+	 * @param  array $atts The attribute values passed in through the shortcode.
+	 * @return string The HTML markup for the donation form.
+	 */
+	public function donation_form_shortcode( $atts ) {
+		if ( isset( $atts['type'] ) && 'select' === $atts['type'] ) {
+			return $this->render_view( '/views/rr-donation-form-select.view.php', $atts );
+		} else {
+			return $this->render_view( '/views/rr-donation-form-buttons.view.php', $atts );
+		}
+	}
+
+	/**
+	 * Register the donation plugin's shortcode's CSS and Javascript files.
+	 */
+	public function register_assets() {
+		wp_register_style(
+			'newsmatch-donation',
+			plugins_url( 'assets/css/donation.css', NMD_PLUGIN_FILE )
+		);
+
+		wp_enqueue_style( 'newsmatch-donation' );
+
+		wp_register_script(
+			'newsmatch-donation',
+			plugins_url( 'assets/js/donation.js', NMD_PLUGIN_FILE ),
+			array( 'jquery' ),
+			null,
+			true
+		);
+
+		wp_enqueue_script( 'newsmatch-donation' );
+	}
+
+	/**
+	 * Get the URL start for the form that we're using, appropriately escaped
+	 *
+	 * @return string The URL
+	 */
+	private function get_url() {
+		if ( get_option( $this->option_prefix . 'url_toggle' ) === 'staging' ) {
```

Comment:
yoda condition, we must use

Reply:
Yoda conditions are for when comparing variables, to prevent accidental variable assignment, but since we're comparing the output of a function to a string, there's no variable to be assigned, right? Or am I completely misinterpreting the use of yoda conditions?

Reply:
no you're right - that's a good point.
If you want to not use yoda condition here, I would check to see what happens if you actually assign get_option() to 'staging', just to be sure

Reply:
```
if ( get_option( $this->option_prefix . 'url_toggle' ) = 'staging' ) {
```
> Fatal error: Can't use function return value in write context in /srv/www/okwatch/htdocs/wp-content/plugins/newsmatch-donate-plugin/classes/class-newsmatchdonation-shortcode.php on line 82

So that looks like it's not going to cause subtle problems if the `===` gets changed in the future; the problems will be blatantly obvious.